examples/tv-casting-app/android/App/app/src/main/jni/com/chip/casting/TvCastingApp.java

```diff
@@ -136,23 +147,126 @@ public void discoverVideoPlayerCommissioners(
       multicastLock.acquire();
 
       nsdManager = (NsdManager) applicationContext.getSystemService(Context.NSD_SERVICE);
+      discoveredPlayers = new ArrayList<>();
       nsdDiscoveryListener =
           new NsdDiscoveryListener(
               nsdManager,
               DISCOVERY_TARGET_SERVICE_TYPE,
               DISCOVERY_TARGET_DEVICE_TYPE_FILTER,
               preCommissionedVideoPlayers,
-              discoverySuccessCallback,
+              new SuccessCallback<DiscoveredNodeData>() {
+                @Override
+                public void handle(DiscoveredNodeData commissioner) {
+                  Log.d(TAG, "Discovered commissioner added " + commissioner);
+                  discoveredPlayers.add(commissioner);
+                  discoverySuccessCallback.handle(commissioner);
+                }
+              },
               discoveryFailureCallback,
               nsdManagerResolverAvailState);
 
       nsdManager.discoverServices(
           DISCOVERY_TARGET_SERVICE_TYPE, NsdManager.PROTOCOL_DNS_SD, nsdDiscoveryListener);
       Log.d(TAG, "TvCastingApp.discoverVideoPlayerCommissioners started");
+
+      this.reportSleepingVideoPlayerCommissionersFuture =
```

Comment:
can you add a comment indicating why we do this?

Reply:
Done.